File: jar/src/main/java/jline/api/mam/Mfq_fluflu_sojourn.java
/**
 * @file Sojourn-time distribution of a fluid/fluid queue
 * @since LINE 3.0
 */
package jline.api.mam;

import jline.lib.butools.queues.FluidQueueSTD;
import jline.util.matrix.Matrix;

public final class Mfq_fluflu_sojourn {
    private Mfq_fluflu_sojourn() {}

    /** Sojourn-time distribution of a fluid queue with fluid-modulated service, as {alpha, A}. */
    public static Matrix[] mfq_fluflu_sojourn(Matrix Qin, Matrix Rin, Matrix Qout, Matrix Rout, boolean srv0stop, boolean transToPH) {
        return FluidQueueSTD.fluFluSTD(Qin, Rin, Qout, Rout, srv0stop, transToPH);
    }
}

File: matlab/src/api/mam/mfq_fluflu_sojourn.m
%{ @file mfq_fluflu_sojourn.m
 %  @brief Sojourn-time distribution of a fluid/fluid queue
 %
 %  @author LINE Development Team
%}

%{
 % @brief Sojourn-time distribution of a fluid queue with fluid-modulated service.
 %
 % @details
 % Thin LINE wrapper around FluFluSTD. Returns the sojourn-time distribution of
 % a fluid queue in which both the arrival and the service processes are
 % Markov-modulated fluid flows, as a matrix-exponential (ME) or phase-type
 % (PH) representation (alpha, A).
 %
 % @par Syntax:
 % @code
 % [alpha,A] = mfq_fluflu_sojourn(Qin,Rin,Qout,Rout,srv0stop)
 % [alpha,A] = mfq_fluflu_sojourn(Qin,Rin,Qout,Rout,srv0stop,transToPH)
 % @endcode
 %
 % @par Parameters:
 % <table>
 % <tr><th>Name<th>Description
 % <tr><td>Qin<td>(Na,Na) generator of the arrival-modulating chain
 % <tr><td>Rin<td>(Na,Na) diagonal arrival fluid-rate matrix
 % <tr><td>Qout<td>(Ns,Ns) generator of the service-modulating chain
 % <tr><td>Rout<td>(Ns,Ns) diagonal service fluid-rate matrix
 % <tr><td>srv0stop<td>true if service stops when the server fluid level hits zero
 % <tr><td>transToPH<td>(Optional) true to return a PH (else ME) representation
 % </table>
 %
 % @par Returns:
 % alpha, A: ME/PH representation of the sojourn-time distribution.
%}
function [alpha,A] = mfq_fluflu_sojourn(Qin,Rin,Qout,Rout,srv0stop,transToPH)
if ~exist('transToPH','var'), transToPH = false; end
[alpha,A] = FluFluSTD(Qin,Rin,Qout,Rout,srv0stop,transToPH);
end
